modules/core/src/main/java/org/apache/ignite/internal/processors/cache/checker/objects/VersionedEntry.java

```diff
@@ -20,89 +20,99 @@
     /**
      *
      */
     private static final long serialVersionUID = 0L;
 
-    /** Value. */
-    private CacheObject val;
+    /** Key. */
+    private VersionedKey key;
 
-    /**
-     * Partition update counter for the moment of read from data store.
-     */
-    private long updateCounter;
-
-    /**
-     * Recheck start time.
-     */
-    private long recheckStartTime;
+    /** Value. */
+    private VersionedValue val;
 
     /**
      * Default constructor.
      */
-    public PartitionDataRow() {
+    public VersionedEntry() {
+        // No-op
     }
 
     /**
      * @param nodeId Node id.
      * @param key Key.
      * @param ver Version.
      * @param val Value.
-     * @param updateCounter Update counter.
+     * @param updateCntr Update counter.
      * @param recheckStartTime Recheck start time.
      */
-    public PartitionDataRow(UUID nodeId, KeyCacheObject key, GridCacheVersion ver, CacheObject val, long updateCounter,
+    public VersionedEntry(UUID nodeId, KeyCacheObject key, GridCacheVersion ver, CacheObject val, long updateCntr,
         long recheckStartTime) {
-        super(nodeId, key, ver);
-        this.val = val;
-        this.updateCounter = updateCounter;
-        this.recheckStartTime = recheckStartTime;
+        this.key = new VersionedKey(nodeId, key, ver);
+        this.val = new VersionedValue(val, ver, updateCntr, recheckStartTime);
     }
 
     /**
      * @return Entry value.
      */
     public CacheObject getVal() {
-        return val;
+        return val.value();
     }
 
     /**
      * @return Partition update counter for the moment of read from data store.
      */
-    public long getUpdateCounter() {
-        return updateCounter;
+    public long getUpdateCntr() {
+        return val.updateCounter();
     }
 
     /**
      * @return Recheck start time.
      */
     public long getRecheckStartTime() {
-        return recheckStartTime;
+        return val.recheckStartTime();
     }
 
     /** {@inheritDoc} */
     @Override protected void writeExternalData(ObjectOutput out) throws IOException {
-        super.writeExternalData(out);
+        out.writeObject(key);
         out.writeObject(val);
-        out.writeLong(updateCounter);
-        out.writeLong(recheckStartTime);
-
     }
 
     /** {@inheritDoc} */
-    @Override protected void readExternalData(byte protoVer, ObjectInput in) throws IOException, ClassNotFoundException {
-        super.readExternalData(protoVer, in);
-        val = (CacheObject)in.readObject();
-        updateCounter = in.readLong();
-        recheckStartTime = in.readLong();
+    @Override protected void readExternalData(byte protoVer,
+        ObjectInput in) throws IOException, ClassNotFoundException {
+        key = (VersionedKey)in.readObject();
+        val = (VersionedValue)in.readObject();
+    }
+
+    /**
+     * @return Key of this entry.
+     */
+    public KeyCacheObject getKey() {
```

Comment:
In GG code style there's no "get" in getters, isn't it?
I don't see this as a must, but some DTO has "gets" and some hasn't. Looks misleading.